### PR TITLE
feat: use multistage builds

### PR DESCRIPTION
**Pull request by fPolic**
**What**
- reorganise Dockerfiles to use multistage build
- optimise the size of (final) running containers
- fix issues where unnecessary things were copied to the context 
- add .dockerignore files

### CODE REVIEW

**Comment by Prn-Ice** [comment]
I'm currently trying this out, I had a few issues.
- The admin and storefront docker files are the same, even the working directory files, is this a mistake?
- Without the `RUN npm install -g gatsby-cli` section on both the admin and storefront, both fail to start
- In the docker-compose.dev.yml, the postgres service is defined without an image

Even after fixing these issues, I still cant get the storefront to start but I'm working on it.

File: docker-compose.dev.yml [hunk 1]
```diff
@@ -2,23 +2,44 @@ version: "3.8"
...
       - redis
     environment:
+      NODE_ENV: development
       DATABASE_URL: postgres://postgres:postgres@postgres:5432/medusa-docker
+      STORE_CORS: http://localhost
       REDIS_URL: redis://cache
-      NODE_ENV: development
       JWT_SECRET: some_jwt_secret
       COOKIE_SECRET: some_cookie_secret
       PORT: 9000
```
Comment (carlos-r-l-rodrigues) [inline docker-compose.dev.yml:20]: You can add this env var `CHOKIDAR_USEPOLLING: true` to make the hot reload work

File: docker-compose.dev.yml [hunk 2]
```diff
@@ -2,23 +2,44 @@ version: "3.8"
...
+      - backend
+    container_name: medusa-admin
+    environment:
+      NODE_OPTIONS: --openssl-legacy-provider
+    ports:
+      - "7000:7000"
 
   postgres:
-    image: postgres:10.4
+    restart: always
```
Comment (carlos-r-l-rodrigues) [inline docker-compose.dev.yml:44]: Missing image

**Comment by pan-sveta** [comment]
I would say that you miss ` as prod` in the backend Dockerfile.

Otherwise, the production compose is failing for me.

**Pull request closed without merging**